helm/cluster-api-provider-azure/templates/rbac.yaml

```diff
@@ -0,0 +1,333 @@
+  - apiGroups:
+      - exp.infrastructure.cluster.x-k8s.io
+    resources:
+      - azuremachinepools
+    verbs:
+      - create
+      - delete
+      - get
+      - list
+      - patch
+      - update
+      - watch
+  - apiGroups:
+      - exp.infrastructure.cluster.x-k8s.io
+    resources:
+      - azuremachinepools/status
+    verbs:
+      - get
+      - patch
+      - update
+  - apiGroups:
+      - exp.infrastructure.cluster.x-k8s.io
+    resources:
+      - azuremanagedclusters
+    verbs:
+      - create
+      - delete
+      - get
+      - list
+      - patch
+      - update
+      - watch
+  - apiGroups:
+      - exp.infrastructure.cluster.x-k8s.io
+    resources:
+      - azuremanagedclusters/status
+    verbs:
+      - get
+      - patch
+      - update
+  - apiGroups:
+      - exp.infrastructure.cluster.x-k8s.io
+    resources:
+      - azuremanagedcontrolplanes
+    verbs:
+      - create
+      - delete
+      - get
+      - list
+      - patch
+      - update
+      - watch
+  - apiGroups:
+      - exp.infrastructure.cluster.x-k8s.io
+    resources:
+      - azuremanagedcontrolplanes/status
+    verbs:
+      - get
+      - patch
+      - update
+  - apiGroups:
+      - exp.infrastructure.cluster.x-k8s.io
+    resources:
+      - azuremanagedmachinepools
+    verbs:
+      - create
+      - delete
+      - get
+      - list
+      - patch
+      - update
+      - watch
+  - apiGroups:
+      - exp.infrastructure.cluster.x-k8s.io
+    resources:
+      - azuremanagedmachinepools/status
+    verbs:
+      - get
+      - patch
+      - update
+  - apiGroups:
+      - infrastructure.cluster.x-k8s.io
+    resources:
+      - azureclusteridentities
+      - azureclusteridentities/status
+    verbs:
+      - create
+      - delete
+      - get
+      - list
+      - patch
+      - update
+      - watch
+  - apiGroups:
+      - infrastructure.cluster.x-k8s.io
+    resources:
+      - azureclusters
+    verbs:
+      - create
+      - delete
+      - get
+      - list
+      - patch
+      - update
+      - watch
+  - apiGroups:
+      - infrastructure.cluster.x-k8s.io
+    resources:
+      - azureclusters/status
+    verbs:
+      - get
+      - patch
+      - update
+  - apiGroups:
+      - infrastructure.cluster.x-k8s.io
+    resources:
+      - azuremachines
+    verbs:
+      - create
+      - delete
+      - get
+      - list
+      - patch
+      - update
+      - watch
+  - apiGroups:
+      - infrastructure.cluster.x-k8s.io
+    resources:
+      - azuremachines/status
+    verbs:
+      - get
+      - patch
+      - update
+  - apiGroups:
+      - infrastructure.cluster.x-k8s.io
+    resources:
+      - azuremachinetemplates
+      - azuremachinetemplates/status
+    verbs:
+      - get
+      - list
+      - watch
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  labels:
+    {{ include "labels.provider" . | nindent 4 }}
+  name: {{ include "resource.default.name" . }}-metrics-reader
```

Comment:
Is this role used anywhere?

Reply:
Checking that, I believe we can remove it.

Reply:
Yeah I think so

Reply:
Will leave it now and we can check it later.